lib/network.py

```diff
@@ -578,7 +582,8 @@ def start_interface(self, server_key):
                 self.print_error("connecting to %s as new interface" % server_key)
                 self.set_status('connecting')
             self.connecting.add(server_key)
-            c = Connection(server_key, self.socket_queue, self.config.path)
+            c = Connection(server_key, self.socket_queue, self.config.path,
+                lambda x: x.bad_certificate.append_weak(self.on_bad_certificate))
```

Comment:
Hmm.. I wonder if it would be more idiomatic here to have the `Connection` function just `append_weak` the supplied callback for you to the  `bad_certificate` event list.... of course this works.. just wondering what would be more idiomatic.  feel free to ignore this comment.

Reply:
Yea, I also wasn't sure which one would be the better choice. I chose this because I feel like this is the more generic option, allowing the caller to customize anything about the connection before starting it.

Reply:
Ok yeah.. leave it. Also the nice thing is reading this code I know immediately what is happening...